openassessment/xblock/static/js/src/studio/oa_edit_assessment.js

```diff
@@ -621,3 +621,104 @@ OpenAssessment.EditExampleBasedAssessmentView.prototype = {
     validationErrors: function() { return []; },
     clearValidationErrors: function() {}
 };
+
+/**
+ Interface for editing staff assessment settings.
+
+ Args:
+ element (DOM element): The DOM element representing this view.
+
+ Returns:
+ OpenAssessment.EditStaffAssessmentView
+
+ **/
+OpenAssessment.EditStaffAssessmentView = function(element) {
+    this.element = element;
+    this.name = "staff-assessment";
+
+    // Configure the toggle checkbox to enable/disable this assessment
+    new OpenAssessment.ToggleControl(
+        $("#include_staff_assessment", this.element),
+        $("#staff_assessment_description", this.element),
+        $("#staff_assessment_description", this.element), //open and closed selectors are the same!
+        new OpenAssessment.Notifier([
+            new OpenAssessment.AssessmentToggleListener()
+        ])
+    ).install();
+};
+
+OpenAssessment.EditStaffAssessmentView.prototype = {
+
+    /**
+     Return a description of the assessment.
+
+     Returns:
+     object literal
+     **/
+    description: function() {
+        return {
+            required: this.isEnabled(),
+        };
+    },
+
+    /**
+     Get or set whether the assessment is enabled.
+
+     Args:
+     isEnabled (boolean, optional): If provided, set the enabled state of the assessment.
+
+     Returns:
+     boolean
+     ***/
+    isEnabled: function(isEnabled) {
```

Comment:
This code is crying out for a base class that the various editors can extend (since it seems like there is a lot of boiler-plate code). However, I reluctantly say "not now".